pilot/pkg/xds/ads.go

```diff
@@ -398,7 +398,17 @@ func (s *DiscoveryServer) initializeProxy(con *Connection) error {
 }
 
 func (s *DiscoveryServer) computeProxyState(proxy *model.Proxy, request *model.PushRequest) {
-	proxy.SetServiceTargets(s.Env.ServiceDiscovery)
+	var shouldResetGateway, shouldResetSidecarScope bool
+	// 1. If request == nil(initiation phase) or request.ConfigsUpdated == nil(global push), set proxy serviceTargets.
+	// 2. otherwise only set when svc update, this is for the case that a service may select the proxy
+	if request == nil || len(request.ConfigsUpdated) == 0 ||
+		model.HasConfigsOfKind(request.ConfigsUpdated, kind.Service) {
```

Comment:
Service can never be set?
```
	if model.HasConfigsOfKind(req.ConfigsUpdated, kind.Service) {
			panic("assertion failed kind.Service can not be set in ConfigKey")
		}
```

Reply:
Hmm, i meant service entry...

Reply:
We always use ServiceEntry from the begining, since some time, we added native k8s resource type into the package. So mis used it